tests/unit/browser/test_cookies.py

```diff
@@ -0,0 +1,190 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutebrowser.browser.cookies"""
+
+from unittest import mock
+
+from PyQt5.QtNetwork import QNetworkCookie
+from PyQt5.QtTest import QSignalSpy
+from PyQt5.QtCore import QUrl
+import pytest
+
+from qutebrowser.browser import cookies
+from qutebrowser.utils import objreg
+from qutebrowser.misc import lineparser
+
+CONFIG_ALL_COOKIES = {'content': {'cookies-accept': 'all'}}
+CONFIG_NEVER_COOKIES = {'content': {'cookies-accept': 'never'}}
+CONFIG_COOKIES_ENABLED = {'content': {'cookies-store': True}}
+
+
+cookie1 = b'foo1=bar; expires=Tue, 01-Jan-2036 08:00:01 GMT'
+cookie2 = b'foo2=bar; expires=Tue, 01-Jan-2036 08:00:01 GMT'
+session_cookie = b'foo3=bar'
+expired_cookie = b'foo4=bar; expires=Sat, 01-Jan-2000 08:00:01 GMT'
+
+
+class LineparserSaveStub(lineparser.BaseLineParser):
+    """A stub for LineParser's save()
+
+    Attributes:
+        data: The data before the write
+        saved: The .data before save()
+    """
+
+    def __init__(self, configdir, fname, *, binary=False, parent=None):
+        super().__init__(configdir, fname, binary=binary, parent=parent)
+        self.saved = []
+        self.data = []
+
+    def save(self):
+        self.saved = self.data
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __getitem__(self, key):
+        return self.data[key]
+
+
+@pytest.yield_fixture
+def fake_save_manager():
+    """Create a mock of save-manager and register it into objreg."""
+    fake_save_manager = mock.Mock()
+    objreg.register('save-manager', fake_save_manager)
+    yield
+    objreg.delete('save-manager')
+
+
+def test_set_cookies_accept(config_stub, qtbot, monkeypatch):
+    """Test setCookiesFromUrl with cookies enabled."""
+    config_stub.data = CONFIG_ALL_COOKIES
+    ram_jar = cookies.RAMCookieJar()
+    cookie = QNetworkCookie(b'foo', b'bar')
+    url = QUrl('http://example.com/')
+    with qtbot.waitSignal(ram_jar.changed, raising=True):
+        assert ram_jar.setCookiesFromUrl([cookie], url)
+
+    # assert the cookies are added correctly
+    all_cookies = ram_jar.cookiesForUrl(url)
+    assert len(all_cookies) == 1
+    saved_cookie = all_cookies[0]
+    expected = cookie.name(), cookie.value()
+    assert saved_cookie.name(), saved_cookie.value() == expected
+
+
+def test_set_cookies_never_accept(config_stub):
+    """Test setCookiesFromUrl when cookies are not accepted."""
+    config_stub.data = CONFIG_NEVER_COOKIES
+    ram_jar = cookies.RAMCookieJar()
+    changed_signal_spy = QSignalSpy(ram_jar.changed)
+
+    assert not ram_jar.setCookiesFromUrl('test', 'test')
+    assert not changed_signal_spy
```

Comment:
Maybe also check there really are no cookies set, i.e. `assert not ram_jar.cookiesForUrl(url)`